commcare_connect/program/helpers.py

```diff
@@ -76,3 +76,55 @@ def get_annotated_managed_opportunity(program: Program):
     )
 
     return managed_opportunities
+
+
+def get_delivery_performance_report(program: Program, start_date, end_date):
+    date_filter = FILTER_FOR_VALID_VISIT_DATE
+
+    if start_date:
+        date_filter &= Q(opportunityaccess__uservisit__visit_date__gte=start_date)
+
+    if end_date:
+        date_filter &= Q(opportunityaccess__uservisit__visit_date__lte=end_date)
+
+    flagged_visits_filter = Q(opportunityaccess__uservisit__flagged=True) & ~Q(
+        opportunityaccess__uservisit__status__in=[
+            VisitValidationStatus.rejected,
+            VisitValidationStatus.approved,
+        ]
+    )
+
+    managed_opportunities = (
+        ManagedOpportunity.objects.filter(program=program)
+        .order_by("start_date")
+        .prefetch_related(
```

Comment:
Same as the other PR.

Reply:
Removed [78fe8cd](https://github.com/dimagi/commcare-connect/pull/402/commits/78fe8cd983e6cb2381c1edd63f73eea1d9b4c2a3)